apps/emqx_resource/src/emqx_resource_buffer_worker.erl

```diff
@@ -1341,9 +1378,9 @@ mark_inflight_items_as_retriable(Data, WorkerMRef) ->
     ok.
 
 %% used to update a batch after dropping expired individual queries.
-update_inflight_item(InflightTID, Ref, NewBatch) ->
+update_inflight_item(InflightTID, Ref, NewBatch, NumExpired) ->
     _ = ets:update_element(InflightTID, Ref, {?ITEM_IDX, NewBatch}),
-    ?tp(buffer_worker_worker_update_inflight_item, #{ref => Ref}),
+    _ = ets:update_counter(InflightTID, ?SIZE_REF, {2, -NumExpired, 0, 0}),
```

Comment:
Add a function guard `when NumExpired > 0`

Reply:
I mean, maybe we should guard just this line?  Otherwise, trying to update an element without any expired items might blow up.

```suggestion
    NumExpired > 0 andalso (_ = ets:update_counter(InflightTID, ?SIZE_REF, {2, -NumExpired, 0, 0})),
```
?